main.js

```diff
@@ -11,28 +11,52 @@ function generateNumbers() {
     return allNumbers;
 }
 
-function displayNumbers(allNumbers) {
-    const numbersOnScreen = allNumbers.length;
+function displayNumbers() {
+    const randomNumbers = generateNumbers();
+    const sortedNumbers = randomNumbers.slice(0).sort((a, b) => a - b);
+    const numbersOnScreen = randomNumbers.length;
+    let selectedOrder = [];
 
-     // using a for loop instead of forEach as the array.pop changes the length over each iteration
-     for (let i = 0; i < numbersOnScreen; i++) {
-        let randNumber = allNumbers.pop();
+    // using a for loop instead of forEach as the array.pop changes the length over each iteration
+    for (let i = 0; i < numbersOnScreen; i++) {
+        let randNumber = randomNumbers.pop();
         let number = document.createElement('div');
 
         number.textContent = randNumber;
         number.setAttribute('class', 'numberBox');
+
         document.querySelector('#play_area').appendChild(number);
     }
+
+    document.querySelectorAll('.numberBox').forEach((element) => {
+        element.addEventListener('click', () => {
+            const { textContent, classList } = element;
+            const value = parseInt(textContent);
+
+            if (!classList.contains('chosen_number')) {
+                if (sortedNumbers[selectedOrder.length] === value) {
```

Comment:
suggest adding some comments in here that explain what this code does, its not currently very readable

Reply:
or make it more readable :)